astropy/coordinates/angles/formats.py

```diff
@@ -319,52 +319,39 @@ def parse(self, angle, unit, debug=False):
         return found_angle, found_unit
 
 
-def _check_hour_range(hrs):
+def _check_hour_range(hrs: float) -> None:
     """
-    Checks that the given value is in the range (-24, 24).
+    Checks that the given value is in the range [-24,24].  If the value
+    is equal to -24 or 24, then a warning is raised.
     """
-    if np.any(np.abs(hrs) == 24.0):
+    if abs(hrs) == 24.0:
```

Comment:
Probably really not worth optimizing this further, but in principle one would want OK hours to go through fastest, so I think the pattern one wants is,
```
if not -24.0 < hrs < 24.0:
    if abs(hrs) > 24.0:
        raise(...)
    warn(...)
```

Reply:
Code for timing the function:
```
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.15.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from astropy.coordinates.angles.formats import _check_hour_range

In [2]: %timeit _check_hour_range(15)
```
Results on current `main`:
```
8.79 µs ± 22.6 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```
With my original patch:
```
79.8 ns ± 0.0881 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```
With your suggestion:
```
59.5 ns ± 0.118 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```
So the difference between my original patch and your suggestion is not large enough to be noticeable when running `pytest astropy/coordinates`, but your suggestion is faster.